cvmfs/fuse_evict.cc

```diff
@@ -190,14 +199,26 @@ void *FuseInvalidator::MainInvalidator(void *data) {
                entry_parent, entry_name.c_str());
       // Can fail, e.g. the entry might be already evicted
 #if CVMFS_USE_LIBFUSE == 2
-      fuse_lowlevel_notify_inval_entry(*reinterpret_cast<struct fuse_chan**>(
-        invalidator->fuse_channel_or_session_),
-        entry_parent, entry_name.GetChars(), entry_name.GetLength());
+      struct fuse_chan* channel_or_session =
+                                    *reinterpret_cast<struct fuse_chan**>(
+                                     invalidator->fuse_channel_or_session_);
 #else
-      fuse_lowlevel_notify_inval_entry(*reinterpret_cast<struct fuse_session**>(
-        invalidator->fuse_channel_or_session_),
+      struct fuse_session* channel_or_session =
+                                  *reinterpret_cast<struct fuse_session**>(
+                                  invalidator->fuse_channel_or_session_);
+#endif
+
+// we do not care if fuse kernel supports expire_entry as if it is
+// not support it will just be handled like a fuse_inval
+#ifdef FUSE_CAP_EXPIRE_ONLY
+      fuse_lowlevel_notify_expire_entry(channel_or_session,
+        entry_parent, entry_name.GetChars(), entry_name.GetLength(),
+        fuse_expire_flags::FUSE_LL_EXPIRE_ONLY);
```

Comment:
I guess this doesn't work with C++03
```suggestion
        FUSE_LL_EXPIRE_ONLY);
```